external_parser/example_joiner.cc

```diff
@@ -11,90 +12,16 @@
 #include <time.h>
 
 // VW headers
-#include "example.h"
 #include "io/logger.h"
 #include "parse_example_json.h"
 #include "parser.h"
-#include "v_array.h"
-
-namespace RewardFunctions {
-float average(const joined_event &event) {
```

Comment:
not for this PR, but just a question:
is there a strong reason why this interface takes joined_event as input but not const <outcome_event>&?

Reply:
I think mostly future-proofing since a more complex reward function might need more information. We could make it more strict and only use `outcome_event` if we see we don't use anything else from `joiend_event` @cheng-tan what do you think?